includes/http/HTTPHandler.hpp

```diff
@@ -1,13 +1,20 @@
 #ifndef HTTP_HANDLER_HPP
 #define HTTP_HANDLER_HPP
 
+#include "http/HTTPParser.hpp"
 #include "http/HTTPRequest.hpp"
+#include "http/HTTPResponse.hpp"
 #include "socket/ServerSocket.hpp"
 #include "utils/smt.hpp"
 
 namespace webserv {
 
 void http_handle(smt::shared_ptr<ServerSocket> sock, int client_fd);
+smt::shared_ptr<HTTPResponse>
+    generate_error_response(int code, smt::shared_ptr<ServerBlock> config);
```

Comment:
looks good, but when possible try to pass shared_ptr by reference. This avoids all of the copy constructors code being ran each time the parameter is passed

Reply:
hadn't though of that, I agree its better that way. I will try to think of it from now on.